traefik/templates/ingressclass.yaml

```diff
@@ -1,11 +1,13 @@
-{{- if and .Values.ingressClass.enabled (semverCompare ">=2.3.0" (default .Chart.AppVersion .Values.image.tag)) -}}
-  {{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/IngressClass" }}
+{{- if .Values.ingressClass.enabled -}}
+  {{- $hasigclass := "traefik.hasingressclass" }}
+  {{- if $hasigclass }}
+    {{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/IngressClass" }}
 apiVersion: networking.k8s.io/v1
-  {{- else if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta1/IngressClass" }}
+    {{- else if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta1/IngressClass" }}
```

Comment:
I believe those extra whitespaces are not necessary.

Reply:
they're not indeed.
Looking at other parts in code, I can see there's some 2 spaces indentation each time we enter a if (daemonset, deployment, ...). Arguably, it may improve readability. Though I don't really mind, we can remove it.